src/components/Card.js

```diff
@@ -29,15 +43,20 @@ export default class Card {
       this._handleImageClick(this);
     });
     this.likeButton.addEventListener("click", this._handleLikeButton);
-    this.deleteButton.addEventListener("click", this._handleDeleteButton);
+    this.deleteButton.addEventListener("click", () => {
+      this._handleDeleteClick(this);
+    });
   }
 
-  _handleDeleteButton = () => {
-    this.cardElement.remove();
-    this.cardElement = null;
-  };
-
   _handleLikeButton = () => {
-    this.likeButton.classList.toggle("card__like-button_active");
+    if (!this.likeButton.classList.contains("card__like-button_active")) {
+      this.likeButton.classList.add("card__like-button_active");
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.

Reply:
Good point, I have corrected this to catch any errors.